public/app/features/manage-dashboards/DashboardImportPage.tsx

```diff
@@ -0,0 +1,178 @@
+import { ImportDashboardOverview } from './components/ImportDashboardOverview';
+import { DashboardFileUpload } from './components/DashboardFileUpload';
+import { fetchGcomDashboard, importDashboardJson } from './state/actions';
+import appEvents from 'app/core/app_events';
+import { getNavModel } from 'app/core/selectors/navModel';
+import { StoreState } from 'app/types';
+
+interface OwnProps {}
+
+interface ConnectedProps {
+  navModel: NavModel;
+  isLoaded: boolean;
+}
+
+interface DispatchProps {
+  fetchGcomDashboard: typeof fetchGcomDashboard;
+  importDashboardJson: typeof importDashboardJson;
+}
+
+type Props = OwnProps & ConnectedProps & DispatchProps;
+
+interface State {
+  gcomDashboard: string;
+  dashboardJson: string;
+  dashboardFile: any;
+  gcomError: string;
+}
+
+const importStyles = stylesFactory(() => {
+  return {
+    option: css`
+      margin-bottom: 32px;
+    `,
+  };
+});
+
+class DashboardImportUnConnected extends PureComponent<Props, State> {
+  state: State = {
+    gcomDashboard: '',
+    dashboardJson: '',
+    dashboardFile: '',
+    gcomError: '',
+  };
+
+  onGcomDashboardChange = (event: FormEvent<HTMLInputElement>) => {
+    this.setState({ gcomDashboard: event.currentTarget.value });
+  };
+
+  onDashboardJsonChange = (event: FormEvent<HTMLTextAreaElement>) => {
+    this.setState({ dashboardJson: event.currentTarget.value });
+  };
+
+  onFileUpload = (event: FormEvent<HTMLInputElement>) => {
+    const { importDashboardJson } = this.props;
+    const file = event.currentTarget.files[0];
+
+    const reader = new FileReader();
+    const readerOnLoad = () => {
+      return (e: any) => {
+        let dashboard: any;
+        try {
+          dashboard = JSON.parse(e.target.result);
+        } catch (error) {
+          appEvents.emit(AppEvents.alertError, ['Import failed', 'JSON -> JS Serialization failed: ' + error.message]);
+          return;
+        }
+        importDashboardJson(dashboard);
+      };
+    };
+    reader.onload = readerOnLoad();
+    reader.readAsText(file);
+  };
+
+  onDashboardJsonLoad = () => {
+    const { dashboardJson } = this.state;
+    const { importDashboardJson } = this.props;
+
+    if (!dashboardJson) {
+      return;
+    }
+    let dashboard: any;
+    try {
+      dashboard = JSON.parse(dashboardJson);
+      importDashboardJson(dashboard);
+    } catch (error) {
+      appEvents.emit(AppEvents.alertError, ['Import failed', 'JSON -> JS Serialization failed: ' + error.message]);
+      return;
+    }
+  };
+
+  getGcomDashboard = () => {
+    const { gcomDashboard } = this.state;
+
+    // From DashboardImportCtrl
+    const match = /(^\d+$)|dashboards\/(\d+)/.exec(gcomDashboard);
+    let dashboardId;
+
+    if (match && match[1]) {
+      dashboardId = match[1];
+    } else if (match && match[2]) {
+      dashboardId = match[2];
+    } else {
+      this.setState({
+        gcomError: 'Could not find dashboard',
+      });
+    }
```

Comment:
This state isn't used anywhere. The grafana.com dashboard loading could be a separate form with async validation applied? You could get rid of the redux things rom this component and only lead isLoaded state:)